Frontend/mvp1

Nafanyan opened the pull request.

Nafanyan committed:
Создал новый проект и очистил его от первоначальных данных

Nafanyan committed:
Восстановил старый проект под MVP1

Nafanyan committed:
Поправки в бэк в соответствии с написание фронта (валидация, возвращаемые значения, контроллеры)

SherlockD commented on Application/Events/Commands/CreateEvent/CreateEventCommandValidator.cs:
```diff
@@ -15,7 +15,7 @@ public CreateEventCommandValidator(IEventRepository eventRepository)
         public async Task<ValidationResult> ValidationAsync(CreateEventCommand command)
         {
-            if (command.Name == null)
+            if (command.Name == null || command.Name == "")
```
String.Empty

SherlockD commented on Application/Users/Queries/GetEvents/GetEventsQueryHandler.cs:
```diff
@@ -18,19 +18,30 @@ public GetEventsQueryHandler(IUserRepository userRepository, IAsyncValidator<Get
+                    getEventsQuery.StartEvent, getEventsQuery.EndEvent);
+                List<GetEventsQueryDto> getEventsQueryDtos = new List<GetEventsQueryDto>();
+                GetEventsQueryDto getEventsQueryDto;
+                foreach (Event e in events)
```
Можно заменить методом из linq, подумай как)

SherlockD replied:
Я имею ввиду вообще конвертацию объектов тут

SherlockD commented on Infrastructure/Data/Users/UserRepository.cs:
```diff
@@ -22,9 +22,12 @@ public async Task<User> GetByIdAsync(long id)
         {
-            return await _dbSetEvent.Where(e => e.UserId == id).ToListAsync();
+            return await _dbSetEvent.Where(e => e.UserId == id)
+                .Where(e => e.StartEvent >= startPeriod)
```
Не знаю как в других местах принятно, но везде где я видел условие в одном Where, если поразмыслить наверное быстрее будет работать. Но в целом не критично, если хочешь поправь, если не хочешь не правь)

SherlockD commented on Web/extranet/web-calendar/src/constants/Montsh.tsx:
```diff
@@ -0,0 +1 @@
+export let months: Array<string> = ['','Январь', 'Февраль', 'Март', 'Апрель', 'Май', 'Июнь', 'Июль', 'Август', 'Сентябрь', 'Октябрь', 'Ноябрь', 'Декабрь'];
```
Почему первый месяц пустой?

Nafanyan replied:
Потому что я работаю с типом Date, в нем январь начинается с 0. А все запросы к БД у меня построены на том, что Январь первый месяц

SherlockD replied:
Выглядит как костыль же, почему запросы с 1 начинаются? Не понимаю

SherlockD commented on Web/extranet/web-calendar/src/css/index.css:
```diff
@@ -0,0 +1,5 @@
+body {
+    background-color:#EAEAEA;
+  }
+  
```
Пустые строки старайся тоже не оставлять

SherlockD commented on Web/extranet/web-calendar/src/models/IValidationResult.tsx:
```diff
@@ -0,0 +1,4 @@
+export interface IValidationResult{
+    isFail: boolean,
+    error: string | null
```
В тайпскрипте вроде null propagating operator тоже должет быть, как в C#

SherlockD commented on Web/extranet/web-calendar/src/models/type/currentDay.tsx:
```diff
@@ -0,0 +1,27 @@
+    year: number;
+    month: number;
+    day: number;
+    nextRendering?: boolean;
```
Вот тут ты его используешь, а почему выше пишешь string | null )))

SherlockD commented on Web/extranet/web-calendar/src/models/type/currentDay.tsx:
```diff
@@ -0,0 +1,27 @@
+    nextRendering?: boolean;
+}
+
+export enum CurrentDayActionType{
```
Для чего эта штука?)

Nafanyan replied:
Вообще чтобы перерендеривать некоторые элементы. Например я удалил событие, закрываю всплывающее окно и и информация о текущем положении дел должна обновиться, для этого я меняю состояние этой переменной, от смены её значений зависит хук useeffect в нужных зависимых объектах. Я переименовал на reRender

SherlockD commented on Web/extranet/web-calendar/src/pages/Calendar/MonthCalendar.tsx:
```diff
@@ -0,0 +1,133 @@
+    const { userId, year, month, nextRendering } = useTypedSelector(state => state.currentDay);
+
+    useEffect(() => {
+        const create2DArray = async () => {
```
Принцип единственной ответственности нарушается. У тебя метод который вроде должен создавать просто массив данных че то тут мутит жесткое, так не должно быть

SherlockD commented on Web/extranet/web-calendar/src/pages/Calendar/MonthCalendar.tsx:
```diff
@@ -0,0 +1,133 @@
+            let startEventStr: string = TimeToStringRequest(new Date(year, monthForCreateDate, 1, 0, 0));
+            let endEventStr: string = TimeToStringRequest(new Date(year, monthForCreateDate, monthDaysCount, 23, 59));
+            let events: IEvent[] = await service.getEvent(userId, startEventStr, endEventStr);
+            console.log(events)
```
логи нужно убирать, после дебага

SherlockD commented on Web/extranet/web-calendar/src/pages/Calendar/MonthCalendar.tsx:
```diff
@@ -0,0 +1,133 @@
+                                                                    <EventInfo startEvent={eventsDay.startEvent} endEvent={eventsDay.endEvent} />
+                                                                    {" " + (eventsDay.name.length > 9 ? eventsDay.name.substring(0,9) + "..." : eventsDay.name + " ")}
+                                                                </Card.Text>}
+
```
Пустая строка

SherlockD replied:
или так должно быть?)

SherlockD commented on Web/extranet/web-calendar/src/pages/Calendar/MonthCalendar.tsx:
```diff
@@ -0,0 +1,133 @@
+                            {week.map(
+                                (day, keyDay) => (
+                                    <th key={keyDay}>
+                                        {new Date(day.arrayEvents[0].startEvent.toString()).getMonth() == month - 1 &&
```
Как это работает?) new Date(day.arrayEvents[0].startEvent.toString()).getMonth() == month - 1 &&, что за && в конце?)

Nafanyan replied:
&& это оператор условия типа if

SherlockD commented on Web/extranet/web-calendar/src/pages/Calendar/WeekCalendar.tsx:
```diff
@@ -0,0 +1,117 @@
+    const { userId, year, month, day, nextRendering } = useTypedSelector(state => state.currentDay);
+
+    useEffect(() => {
+        const create2DArray = async () => {
```
в нескольких местах используется? Если так можно вынести куда нибудь. Ну и метод конечно разбить на несколько

Nafanyan replied:
Не совсем, названия похожи потому что выполняют одну задачу, но каждый со своей спецификой. 

В режиме месяца формируется запрос к АПИ с данными начала включительно 00:00 1го дня месяца и конца включительно 23:59  последнего дня месяца.

В режиме недели рассчитываются даты начала недели и конца недели.

В режиме дня просто формируются включительно с 00:00 до включительно 23:59 одного дня.

Nafanyan committed:
Внес правки в соответствии с замечаниями:
1. Упростил конвертацию объектов в GetEventsQueryHandler
2. Упростил выражения с обращению к Api в React проекте, убрал лишние детали

Nafanyan committed:
Исправил ошибку с валидацией при которой события имели возможность накладываться друг на друга

SherlockD commented on Application/Events/Commands/UpdateEvent/UpdateEventCommandValidator.cs:
```diff
@@ -15,7 +15,7 @@ public UpdateEventCommandValidator(IEventRepository eventRepository)
         public async Task<ValidationResult> ValidationAsync(UpdateEventCommand command)
         {
-            if (command.Name == null)
+            if (command.Name == null || command.Name == string.Empty)
```
Если string как объект, то с большой буквы

SherlockD commented on Application/Users/Queries/GetEvents/GetEventsQueryHandler.cs:
```diff
@@ -18,19 +18,25 @@ public GetEventsQueryHandler(IUserRepository userRepository, IAsyncValidator<Get
+                IReadOnlyList<Event> events = await _userRepository.GetEventsAsync(getEventsQuery.UserId,
+                    getEventsQuery.StartEvent, getEventsQuery.EndEvent);
+
+                List<GetEventsQueryDto> getEventsQueryDtos = events.Select(e => new GetEventsQueryDto
```
во, молодец

Nafanyan committed:
исправил костыль с месяцами, сделал тип данных IDay, который решает проблему с созданием ненужных пустых событий

Nafanyan committed:
небольшие правки с наименованием переменных

Nafanyan committed:
string => String
